src/haddock/gear/validations.py

```diff
@@ -12,3 +28,51 @@ def v_rundir(rundir: FilePath) -> None:
             r"[a-zA-Z0-9._-/\] characters."
             )
         raise ConfigurationError(emsg)
+
+
+def validate_yaml_params_scheme(yaml_fpath: str) -> None:
+    """Validate a default.yaml file parameters schemes.
+
+    Parameters
+    ----------
+    yaml_fpath : str
+        Path to the default.yaml file
+    """
+    ycfg = read_from_yaml(yaml_fpath)
+    allowed_expert_levels = config_expert_levels + ("all", _hidden_level)
+    # Loop over parameters
+    for param, parameters in ycfg.items():
+        for base_subparam in YAML_PARAMETER_BASE_SCHEME:
+            # Check that this parameter contain the basic subparameters
+            assert base_subparam in parameters.keys(), f"Parameter '{param}' is missing base subparam '{base_subparam}' (in {yaml_fpath})."  # noqa : E501
+            param_value = parameters[base_subparam]
+            if base_subparam == "explevel":
+                # Check if expert level is ok
+                assert parameters[base_subparam] in allowed_expert_levels, f"Parameter '{param}' do not contain appropriate expert level (in {yaml_fpath})"  # noqa : E501
+            
+            elif base_subparam == "type":
+                # Check if known type
+                assert param_value in TYPE_SPECIFIC_SCHEMES.keys(), f"Parameter {param} contain an unknown parameter type (in {yaml_fpath})"  # noqa : E501
+                # Skip subsequent checks in case of type: dict
+                if param_value == "dict":
+                    break
```

Comment:
Why this skip here?

Reply:
Parameters of type `dict` do not have default value.